zerver/lib/digest.py

```diff
@@ -214,6 +225,13 @@ def get_hot_topics(
             hot_topics.append(topic)
         if len(hot_topics) == MAX_HOT_TOPICS_TO_BE_INCLUDED_IN_DIGEST:
             break
+    if use_new_ranking_system:
+        if message_weight and sender_weight is not None:
```

Comment:
This conditional is weird and likely doesn't behave as expected.

Reply:
I should change it.